### PR TITLE
Tests: update checkpointing test to single precision

### PR DESCRIPTION
**Pull request by navjotk**
Addresses #1399 by changing the test to use single precision - this now fails with dynamic openmp scheduling (expected). So this PR disables dynamic scheduling for the operator in the test to make it pass.

### CODE REVIEW

File: tests/test_gradient.py [hunk 1]
```diff
@@ -14,24 +14,14 @@ class TestGradient(object):
...
+        This test ensures that the FWI gradient computed with checkpointing matches
+        the one without checkpointing. Note that this test fails with dynamic openmp
+        scheduling enabled so this test disables it.
         """
         spacing = tuple(10. for _ in shape)
-        wave = setup(shape=shape, spacing=spacing, dtype=np.float64,
+        wave = setup(shape=shape, spacing=spacing, dtype=np.float32,
                      kernel=kernel, space_order=space_order,
-                     nbl=40)
+                     nbl=40, opt=('noop', {'openmp': True, 'par-dynamic-work': 1000}))
```
[the remark this answers is not in the document]
Comment (navjotk) [reply tests/test_gradient.py:24]: Good spot there. I had not tried that before. Having tried it now - no it doesn't work. Also `('noop')` (i.e. dynamic scheduling enabled with no optimisations) doesn't work either.

[the remark this answers is not in the document]
Comment (mloubout) [reply tests/test_gradient.py:24]: Hum ok, interesting, to think about, non=-blocking so GTG

**Comment by ggorman** [comment]
I dug into this...the very short version is that I've by changing a single line I can go from a fail to a pass.
Line generated by opt=factorize for which test fails:
```C
grad[x + 1][y + 1] += -(-2.0F*u[time][x + 4][y + 4] + u[time - 1][x + 4][y + 4] + u[time + 1][x + 4][y + 4])*v[t0][x + 4][y + 4]/pow(dt, 2);```
Line generated by opt=noop for which test passes:
```C
grad[x + 1][y + 1] += -(-2.0F*u[time][x + 4][y + 4]/pow(dt, 2) + u[time - 1][x + 4][y + 4]/pow(dt, 2) + u[time + 1][x + 4][y + 4]/pow(dt, 2))*v[t0][x + 4][y + 4];```
Go figure!

**Comment by ggorman** [comment]
I'm closing this issue because it turned out to be rabbit-hole into roundoff non-associativity of floating-point addition madness (and another branch indicates there is also an issue with -funsafe-math-optimizations).

**Pull request closed without merging**